openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -2298,137 +2299,225 @@ def _detect_undefined_stereo(cls, rdmol, err_msg_prefix="", raise_warning=False)
 
     @staticmethod
-    def _flip_rdbond_direction(rdbond, paired_rdbonds):
-        """Flip the rdbond and all those paired to it.
+    def _constrain_end_directions(
+        *values: int, bond_indices: List[int], flip_direction: Dict[int, bool]
+    ) -> bool:
+        """A constraint applied when mapping global E/Z stereochemistry into local RDKit
+        bond directions that ensures that the 'left' bonds point in opposite directions
+        (i.e. one has to be up and one has to be down) and likewise for the 'right' bonds
+        """
+        # Account for bond "direction" using flip_directions dict, see more thorough comment
+        # in _constrain_rank for details.
+        unique_values = [
+            (value if not flip_direction[i] else 1 - value)
+            for i, value in zip(bond_indices, values)
+        ]
+        unique_values_set = set(unique_values)
+        print(unique_values_set)
+        return len(unique_values_set) == len(values)
 
-        Parameters
-        ----------
-        rdbond : rdkit.Chem.Bond
-            The Bond whose direction needs to be flipped.
-        paired_rdbonds : Dict[Tuple[int], List[rdkit.Chem.Bond]]
-            Maps bond atom indices that are assigned a bond direction to
-            the bonds on the other side of the double bond.
+    @staticmethod
+    def _constrain_rank(
+        *values: int,
+        bond_indices: List[int],
+        flip_direction: Dict[int, bool],
+        expected_stereo: str,
+    ) -> bool:
+        """A constraint applied when mapping global E/Z stereochemistry into local RDKit
+        bond directions that ensures that the 'left' bond with the highest CIP rank
+        and the 'right' bond with the highest CIP rank point either in the same direction
+        if Z stereo or opposite directions if E.
         """
-        from rdkit import Chem
+        # The "value" for each bond is ultimately set to either 0 (down) or 1 (up).
+        # However, we also need to know the "direction" of the bond to make sense
+        # of this - Is it coming FROM, or going TO this double bond while going down or up?
+        # This information is in the flipped_values dict. This code assumes that the
+        # "normal" case is when the neighboring bonds are coming FROM the double bond,
+        # and where that's not true, the following line switches the
+        # meaning of "down" and "up" to give us the desired meaning.
+        flipped_values = [
+            (value if not flip_direction[i] else 1 - value)
+            for i, value in zip(bond_indices, values)
+        ]
 
-        # The function assumes that all bonds are either up or down.
-        supported_directions = {Chem.BondDir.ENDUPRIGHT, Chem.BondDir.ENDDOWNRIGHT}
+        # Test for equality of items in flipped_values by turning it into a set and
+        # counting how many values remain.
+        flipped_values_set = set(flipped_values)
+        print(flipped_values_set)
```

Comment:
```suggestion
```

Reply:
🤦 Nice catch. I tried to find these at the end of the day before I committed but couldn't find them, and I assumed I'd removed them already. Maybe time to see an optometrist.